src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -596,6 +647,18 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
         if (!oldFiber) {
           oldFiber = nextOldFiber;
         }
+        if (__DEV__) {
+          if (
+            knownKeysInDev != null &&
+            newChild != null &&
+            typeof newChild.key === 'string'
```

Comment:
Yea, is there anyway you can just avoid adding it by doing the check below this condition instead?